tools/run-clangcldocker.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -e
+COMMAND=$*
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+MAINSOURCE=$SCRIPTPATH/..
+ALL_CROARING_FILES=$(cd $MAINSOURCE && git ls-tree --full-tree --name-only -r HEAD | grep -e ".*\.\(c\|h\|cc\|cpp\|hh\)\$" | grep -vFf clang-format-ignore.txt)
+
+if clang-format-17 --version  2>/dev/null | grep -qF 'version 17.'; then
+  cd $MAINSOURCE; clang-format-17 --style=file --verbose -i "$@" $ALL_CROARING_FILES
+  exit 0
```

Comment:
It would probably be better to `exit $?` so we keep the return value, same below

Reply:
We have

```
set -e
```

So my expectation is that if the command is in error, then we will never make it to `exit 0`. Am I wrong?

Reply:
Ah, yeah, that should cover it, yeah.